### ISSUE TITLE
Feature request: 32 bits per sample

### ISSUE THREAD

**Issue by ryn1x** [image-rs/image-tiff#19]
Would you be able to add support for 32bit float Grayscale Tiffs?

**Comment by Nimmerwoner** [comment]
Since image-tiff supports `u32` nowadays, I managed to read my float tiffs like this:
```
let file_path: &Path = ...;
let file = File::open(file_path).unwrap();
let mut tiff = tiff::decoder::Decoder::new(file).unwrap();
match tiff.read_image().unwrap() {
    ...,
    tiff::decoder::DecodingResult::U32(values) => {
        let f32_values = values
            .into_iter()
            .map(|integer| unsafe { std::mem::transmute::<u32, f32>(integer) })
            .collect();

        Some(vec_to_ndarray(f32_values))
    }
```

**Comment by fintelia** [comment]
Resolved by #70
